readme.md

```diff
@@ -164,20 +164,47 @@ Note:
 
 For more details please refer [these](https://github.com/Azure/autorest/tree/master/docs/generate) docs to generate code from your OpenAPI definition using AutoRest.
 
-## Debugging 
+## Debugging
 
-There are two entry points for debugging autorest.csharp:
+### Debugging projects inside the `autorest.csharp` repo
 
-- Opening AutoRest.CSharp.sln in Visual Studio will contain a debugging profile for each sample and test. These only run the "back half", using the yaml file generated by autorest.core as input.
-- ./Generate.ps1 will execute the entire pipeline by invoking autorest and including autorest.csharp as a plugin. Add:
+There are two entry points for debugging the generator:
 
-```
+- Opening `AutoRest.CSharp.sln` in Visual Studio will contain a debugging profile for each sample and test. These only run the "back half", using the yaml file generated by `autorest.modelerfour` as input.
+- `./Generate.ps1` will execute the entire pipeline by invoking autorest and including autorest.csharp as a plugin. Add:
+
+```yaml
 csharpgen:
   attach: true
 ```
 
 to the autorest configuration to attach a debugger to the plugin process.
 
+### Debugging projects outside the `autorest.csharp` repo
+
+There are two kinds of SDK libraries, one of them is using swaggers as input, the other kind is using TypeSpec files as input.
+
+To debug the generator in a project that takes swaggers as input, you need to add
+```yml
+csharpgen:
+  attach: true
+```
+to the autorest configuration to attach a debugger to the plugin process.
+
+To debug `autorest.csharp` in a project that takes TypeSpec files as input, you need to add
+```
+--option @azure-tools/typespec-csharp.csharpGeneratorPath=/absolute/path/to/AutoRest.CSharp.dll --option @azure-tools/typespec-csharp.debug=true
+```
+to attach a debugger to the plugin process.
+
+Usually you should find the `AutoRest.CSharp.dll` file here: `[Root of autorest.csharp repo]/artifacts/bin/AutoRest.CSharp/Debug/net6.0/AutoRest.CSharp.dll`.
```

Comment:
Can we move this line till the end of this section and clean up the command for TypeSpec project

csharpGeneratorPath=/absolute/path/to/artifacts/AutoRest.CSharp.dll%3Bdebug=true